_posts/2023-11-10-automated-test-generation-chess-puzzles-with-dafny.markdown

```diff
@@ -0,0 +1,288 @@
+---
+layout: post
+title:  "Automated Test Generation: Chess Puzzles with Dafny"
+date:   2023-11-10 11:00:00 -0500
+author: Sasha Fedchin
+---
+Given all the incredible examples that showcase how powerful Dafny is (did you know that part of Dafny is written in Dafny?), you might think that verification can fully replace testing. And in many cases it does! However if, like a number of industrial Dafny users, you want to integrate Dafny code with existing codebases, you will face challenges that verification alone might not solve:
+
+- Perhaps your code is actually a reference implementation for another program and you want to test the equivalence between the two.
+- You might be calling out to external libraries, which are not themselves verified, and would like to test them by calling your Dafny code.
+- You have written high-level specifications for your code and are about to embark on a proof but want to have some initial assurance that your yet-to-be-verified implementation does conform to these specifications. 
```

Comment:
> You have written high-level specifications

What makes a specification high-level? 

> specifications

I would make this singular

> You have written high-level specifications for your code

Is a specification not code?

Reply:
I have changed this line to "You have written a specification for your program..." - I think this should make the phrasing more precise!